From 510a8e3004a5cea4a8f673413a2a31af628d62cc Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Tue, 11 Jan 2011 10:13:53 +0800
Subject: [PATCH 204/435] ENGR00137783-1 MXC V4L2 capture:Add several ioctrl
 support

This patch supports to test VIDIOC_ENUM_FMT,
VIDIOC_ENUM_FRAMESIZES and VIDIOC_DBG_G_CHIP_IDENT ioctrls.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_capture.c | 33 ++++++++++++++++++++++++++++++++-
 1 file changed, 32 insertions(+), 1 deletion(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_capture.c b/test/mxc_v4l2_test/mxc_v4l2_capture.c
index 62d08f1..9a8dc2a 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_capture.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_capture.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2010 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2004-2011 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -141,6 +141,9 @@ int v4l_capture_setup(void)
 	struct v4l2_crop crop;
         int fd_v4l = 0;
 		struct v4l2_mxc_offset off;
+	struct v4l2_dbg_chip_ident chip;
+	struct v4l2_frmsizeenum fsize;
+	struct v4l2_fmtdesc ffmt;
 
         if ((fd_v4l = open(v4l_device, O_RDWR, 0)) < 0)
         {
@@ -148,6 +151,34 @@ int v4l_capture_setup(void)
                 return 0;
         }
 
+	if (ioctl(fd_v4l, VIDIOC_DBG_G_CHIP_IDENT, &chip))
+	{
+                printf("VIDIOC_DBG_G_CHIP_IDENT failed.\n");
+                return -1;
+	}
+	printf("sensor chip is %s\n", chip.match.name);
+
+	fsize.index = g_capture_mode;
+	if (ioctl(fd_v4l, VIDIOC_ENUM_FRAMESIZES, &fsize) < 0)
+	{
+	        printf("VIDIOC_ENUM_FRAMESIZES failed\n");
+	        return -1;
+	}
+
+	printf("sensor frame size is %dx%d\n", fsize.discrete.width,
+					       fsize.discrete.height);
+
+	ffmt.index = g_capture_mode;
+	if (ioctl(fd_v4l, VIDIOC_ENUM_FMT, &ffmt) < 0)
+	{
+	        printf("VIDIOC_ENUM_FMT failed\n");
+	        return -1;
+	}
+	if (ffmt.pixelformat == V4L2_PIX_FMT_YUYV)
+		printf("sensor frame format is YUYV\n");
+	else if (ffmt.pixelformat == V4L2_PIX_FMT_UYVY)
+		printf("sensor frame format is UYVY\n");
+
 	parm.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 	parm.parm.capture.timeperframe.numerator = 1;
 	parm.parm.capture.timeperframe.denominator = g_camera_framerate;
-- 
1.8.0

